mysys/my_getopt.c

```diff
@@ -139,6 +139,19 @@ double getopt_ulonglong2double(ulonglong v)
   return u.dbl;
 }
 
+/**
+  Sets the error code returned by handle_options when handling an invalid
+  variable/option.  The error code is determined by the first invalid
+  configuration parsed.
+*/
+static void set_ho_error(int *ho_error, int err)
+{
+  if (!*ho_error && err)
+    *ho_error= err;
+}
+
+#define SET_HO_ERROR_AND_CONTINUE(e) { set_ho_error(&ho_error, (e)); continue; }
```

Comment:
it'd be simpler without `set_ho_error()` function, you don't use it outside of the macro anyway. And it'd be even simpler if you remove the check for `err!=0` (it cannot happen anyway, one doesn't `continue` without an error), and if you return the last error, not the first — it doesn't really matter what error to return, right?

Then it'll be just
```c
#define SET_HO_ERROR_AND_CONTINUE(e) { ho_error=(e); continue; }
```
10x less code :)

Reply:
Agreed, it doesn't matter if we return the first or the last error code.

I removed the checks and `set_ho_error()`.